logbook/handlers.py

```diff
@@ -642,6 +648,75 @@ def ensure_stream_is_open(self):
             self._open()
 
 
+class CompressionHandler(FileHandler):
+    def __init__(self, filename, mode='ab', encoding=None, level=NOTSET,
+                 format_string=None, delay=False, filter=None, bubble=False,
+                 compression_window_size=4*1024**2, compression_quality=0):
+        self._compression_quality = compression_quality
+        self._bytes_written_since_flush = 0
+        self._compression_window_size = compression_window_size
+        FileHandler.__init__(self, filename, mode, encoding, level, format_string, delay, filter, bubble)
+
+    def should_flush(self):
+        return self._bytes_written_since_flush >= self._compression_window_size
+
+    def write(self, item):
+        self.ensure_stream_is_open()
+        if isinstance(item, str):
+            item = item.encode(encoding=self.encoding)
+        self._write(item)
+        self._bytes_written_since_flush += len(item)
```

Comment:
the `_bytes_written` logic should be pushed down to GzipCompressionHandler - where the Brotli one should handle it itself